pool/rate_limiting_average.go

```diff
@@ -181,6 +191,22 @@ func (p *ewmaRateLimiting) runJob(ctx context.Context, j amboy.Job) time.Duratio
 	}
 	j.UpdateTimeInfo(ti)
 
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithCancel(ctx)
+	func() {
```

Comment:
Did you intend to run this in a separate thread?

Reply:
no it's just here for a scope so that we can defer unlocking the mutex.

Reply:
Could you make it a named function instead of an anonymous one?